pkg/services/serviceaccounts/database/database.go

```diff
@@ -82,71 +87,91 @@ func (s *ServiceAccountsStoreImpl) DeleteServiceAccount(ctx context.Context, org
 	})
 }
 
-func (s *ServiceAccountsStoreImpl) UpgradeServiceAccounts(ctx context.Context) error {
-	basicKeys := s.sqlStore.GetAllOrgsAPIKeys(ctx)
+func (s *ServiceAccountsStoreImpl) GetAPIKeysMigrationStatus(ctx context.Context, orgID int64) (status *serviceaccounts.APIKeysMigrationStatus, err error) {
```

Comment:
this way limits us to only migrate once? 🤔 

is this my understanding that, in the case where users or some api keys do not get migrated, we disable the `migrateAll` button